jerry-ext/module/module.c

```diff
@@ -167,26 +167,38 @@ static const jerry_char_t *on_resolve_absent = (jerry_char_t *) "Module on_resol
  * section and loads one that matches the requested name, caching the result for subsequent requests using the context
  * data mechanism.
  */
-bool
-jerryx_module_native_resolver (const jerry_char_t *name, /**< name of the module */
-                               jerry_value_t *result) /**< [out] where to put the resulting module instance */
+static bool
+jerryx_resolve_native_module (const jerry_value_t canonical_name, /**< canonical name of the module */
+                              jerry_value_t *result) /**< [out] where to put the resulting module instance */
 {
   const jerryx_native_module_t *module_p = NULL;
 
+  jerry_size_t name_size = jerry_get_utf8_string_size (canonical_name);
```

Comment:
Shall we call tostring on the name? Or check if it is a string.

Reply:
We shouldn't need to. This function should not be called directly, even though a pointer to it is available. It should only be called by `jerryx_module_resolve ()` and it should only ever receive a `jerry_value_t` stored in the array of canonical module names created by `jerryx_module_resolve ()`. In fact, this function is not even documented. Only the structure `jerryx_module_native_resolver` is documented, and only the availability of it as a global data symbol is documented, not its contents.